src/ui/public/agg_types/metrics/top_hit.js

```diff
@@ -41,12 +41,7 @@ export const topHitMetricAgg = new MetricAggType({
     {
       name: 'field',
       onlyAggregatable: false,
-      filterFieldTypes: function (vis, value) {
-        if (vis.type.name === 'table' || vis.type.name === 'metric') {
-          return true;
-        }
-        return value === 'number';
-      },
+      filterFieldTypes: '*',
```

Comment:
This change looks like top_hit is now different.  Before, you could only see number fields if the vis wasn't a table or a metric.

Reply:
addressed in https://github.com/elastic/kibana/pull/22756